stacker/hooks/aws_lambda.py

```diff
@@ -61,8 +62,32 @@ def _zip_files(files, root):
 
     contents = zip_data.getvalue()
     zip_data.close()
+    content_hash = _calculate_hash(files, root)
 
-    return contents
+    return contents, content_hash
+
+
+def _calculate_hash(files, root):
+    """ Returns a hash of all of the given files at the given root.
+
+    Args:
+        files (list[str]): file names to include in the hash calculation,
+            relative to ``root``.
+        root (str): base directory to analyze files in.
+
+    Returns:
+        str: A hash of the hashes of the given files.
+    """
+    file_hash = hashlib.md5()
+    for fname in files:
```

Comment:
Do you know if [_find_files](https://github.com/remind101/stacker/blob/ef299e02aaf9e770d55b102092a401c812568fe9/stacker/hooks/aws_lambda.py#L93) returns a sorted list? Should probably perform a sort on the files before iterating through them, since non-deterministic ordering will screw up the hash.